src/bandwidth_limit.cpp

```diff
@@ -61,16 +61,12 @@ namespace libtorrent {
 	{
 		if (m_limit == 0) return;
 
+		m_quota_left += (m_limit * dt_milliseconds + 500) / 1000;
+		if (m_quota_left / 3 > m_limit) m_quota_left = m_limit * 3;
+
 		// avoid integer overflow
-		if (m_limit >= std::numeric_limits<int>::max() / dt_milliseconds)
-		{
-			m_quota_left = std::numeric_limits<int>::max();
-		}
-		else
-		{
-			m_quota_left += (m_limit * dt_milliseconds + 500) / 1000;
-			if (m_quota_left / 3 > m_limit) m_quota_left = m_limit * 3;
-		}
+		m_quota_left = std::min(m_quota_left, std::int64_t(std::numeric_limits<int>::max()));
```

Comment:
I'm not sure how you intended this to work. but the overflow would happen on line 64, right? It doesn't matter what you do *after* undefined behavior is triggered, you need to avoid it from happening in the first place